S/StableHLO/build_tarballs.jl

```diff
@@ -0,0 +1,111 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+const YGGDRASIL_DIR = "../.."
+include(joinpath(YGGDRASIL_DIR, "fancy_toys.jl"))
+include(joinpath(YGGDRASIL_DIR, "platforms", "llvm.jl"))
+
+name = "StableHLO"
+version = v"0.14.6"
+
+sources = [
+    GitSource("https://github.com/openxla/stablehlo.git", "8816d0581d9a5fb7d212affef858e991a349ad6b"),
+]
+
+llvm_versions = [v"17.0.6+0"]
+
+script = raw"""
+cd $WORKSPACE/srcdir/stablehlo
+
+# need to run mlir-tblgen and mlir-pdll on the host
+rm ${prefix}/bin/mlir-tblgen ${prefix}/bin/mlir-pdll
+ln -s ${host_prefix}/bin/mlir-tblgen ${prefix}/bin/mlir-tblgen
+ln -s ${host_prefix}/bin/mlir-pdll ${prefix}/bin/mlir-pdll
```

Comment:
Why do you need to do this?  Executables in `${host_bindir}` have higher priority than those in `${bindir}`: https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/a4ef056f979007f806ff3980dc6e592f9dd4dd1b/src/Runner.jl#L1208-L1211 (BTW, it's better to use the more specific env vars called `bindir`)

Reply:
The reason is that CMake loads `AddMLIR.cmake` which points to the target installation of MLIR. I need the target installation of MLIR for linking, but I also need `mlir-tblgen` and `mlir-pdll` to run on the host (which are run when calling `ninja build`). This is the solution I came up with.